src/General/Tag/Tag.tsx

```diff
@@ -55,22 +55,29 @@ const Tag: React.FunctionComponent<Props> = ({
 };
 
 interface BasicTagProps {
+  /** Set Tag into block style. */
   block?: boolean;
+  /** Set Tag into outline style. */
   outline?: boolean;
 }
 
 interface ButtonTagProps {
   icon?: React.ReactNode;
 }
 
+export type TagActionType = 'add' | 'reset';
 interface ActionTagProps {
-  action?: string;
+  /** Set the Action Tag's type. (action must be chosen as variant props value) */
+  action?: TagActionType;
+  /** A function triggered when icon or action tag is clicked */
   onClick?(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
 }
 
-interface Props extends BasicTagProps, ButtonTagProps, ActionTagProps {
+export type TagVariantType = 'button' | 'action';
+export interface Props extends BasicTagProps, ButtonTagProps, ActionTagProps {
   children: React.ReactNode;
-  variant?: string;
+  /** Set the Tag's type. */
```

Comment:
I think the prop name is clear enough, better to use this space to explain what the values do:
```suggestion
  /** <code>button</code> tags usually show a single piece of information, while <code>action</code> tags are elements in a list (e.g. a list of selected options) */
```